### PR TITLE
Add customer note

### PR DESCRIPTION
**Pull request by dumega**
https://app.asana.com/0/436430054069245/775543293835704/f

So if the merchant enable special instruction on bolt modal and the customer fill in it, then we save it into order customer note. and then show the info on both backend and frontside

### CODE REVIEW

File: app/code/community/Bolt/Boltpay/Model/Observer.php [hunk 1]
```diff
@@ -237,38 +237,84 @@ protected function _addMagentoOrderIdToMessage($incrementId)
...
+            $observer->getTransport()->setHtml($customHtml . $optionsHtml);
+        }
+        elseif( get_class($optionsBlock) === Mage::getConfig()->getBlockClassName('sales/order_info') ){
+            $customInfoBlock = Mage::app()->getLayout()->createBlock('boltpay/sales_order_view_note');
+            $optionsHtml = $observer->getTransport()->getHtml();
+            $customHtml  = $customInfoBlock->toHtml(); 
+            $observer->getTransport()->setHtml(  $optionsHtml.$customHtml );
+        }
+        else{
+            return;
```
Comment (mccottry) [inline app/code/community/Bolt/Boltpay/Model/Observer.php]: Currently, this code breaks my admin form.  The order info box is simply not displayed.  I have not gone very deep into trouble shooting, but I would guess that we are reaching your third condition here and nothing is being returned.

I commented above that I think that this code is unnecessary and possibly undesired.  Regardless, something is happening here that prevents the standard order_info block from rendering when I try to use this code in my test environment.  I haven't looked into this extensively, but I can see that no html is rendered at the time of the order info block and execution seems to stop as if there is an un-handled error.

image

**Comment by daisy1754** [comment]
Sorry I should have clarified earlier. At high level my thinking was "if magento has default field to store this information let's store it". I don't think we need to make a lots of development for customer note if default magento doesn't have any UI to show it.

**Comment by dumega** [comment]
@daisy1754  @mccottry 
The magento 1.9 does have such a field to store customer note, but to be honest, I am not quite sure about its purpose, usually it is not shown neither in the frontside nor in the backend by default, we barely use this field before.

and since it is not a type of comment, I do not think it is a good idea to add the content of it into the list of comments, that's why I create a new block in the order view of admin panel, or it can be ignored easily.

@mccottry I think this customization break the form of order creation it is because the order object does not exist at that time and result in an error. my bad, I only concern about the UI to show the content, and ignore the creation process. 

@daisy1754 well, if now you think this is unnecessary, we can close this PR. do we ?

File: app/code/community/Bolt/Boltpay/Model/Observer.php [hunk 2]
```diff
@@ -237,38 +237,84 @@ protected function _addMagentoOrderIdToMessage($incrementId)
...
+        catch (Exception $e) {
+            Mage::helper('boltpay/bugsnag')->notifyException($e);
         }
 
-        $order->setState(Bolt_Boltpay_Model_Payment::transactionStatusToOrderStatus($transaction->status), true, $msg)
-            ->save();
+    }
+    
+    /**
+    * Prepend an additional order info box to the gift options block.
```
[the remark this answers is not in the document]
Comment (dumega) [reply app/code/community/Bolt/Boltpay/Model/Observer.php]: well, magento 1 usually does not provide a way for the customer to add comment when place an order, and if the merchant want to do that, they have to use some other 3rd-party plugins.

Then if we add this note and show it in the comments list, we also have to notify the merchant to check the comments carefully to see if there is any customer note, cause this behavior break the default work flow somehow why not to create a new block to let the merchant find it easily? just my thought for the note.

**Pull request closed without merging**